lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -82,80 +130,94 @@ def is_active(self) -> bool:
             logger.error("Error while checking Vault health status: %s", e)
             return False
 
-    def is_api_available(self) -> bool:
-        """Return whether Vault is available."""
-        try:
-            self._client.sys.read_health_status(standby_ok=True)
-            return True
-        except (VaultError, RequestException) as e:
-            logger.error("Error while checking Vault health status: %s", e)
-            return False
-
-    def unseal(self, unseal_keys: List[str]) -> None:
-        """Unseal Vault."""
-        for unseal_key in unseal_keys:
-            self._client.sys.submit_unseal_key(unseal_key)
-        logger.info("Vault is unsealed")
+    def enable_audit_device(
+        self, device_type: Literal["file", "syslog", "socket"], path: str
```

Comment:
Instead of using a `Literal`, what do you think about creating an enum?

Reply:
I considered using an enum, but felt that it was too wordy and the string conversions are a bit implicit, and the Literal type had the ergonomics to use this function well enough. I do not have a strong opinion either way.

Reply:
Changed.

Reply:
Funnily enough, the type hint for the enum is `(property) value: Literal['file', 'syslog', 'socket'`